docs/cli/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,133 @@
+# Ingesting and Utilizing RAG Artifacts Design Proposal
+
+**Author**: Daniele Martinoli
+
+**Version**: 0.1 (intentionally "naked" until we agree on the high-level design)
+
+## 1. Introduction
+Goal is to describe how we can extend the InstructLab CLI to include options to ingest a local RAG from pre-processed customer documentation, that we assume to be generated from docling output.
+
+References:<br/>
+**They will be removed later as they can't stay in a public repo**
+* [RAG Artifacts with RHEL AI](https://docs.google.com/document/d/1nALKAawvvNKlTnbeQIXmpIXbo2Lxm2vIw1euxW96TgE)
+* [Productization RHEL AI RAG Atifacts](https://docs.google.com/document/d/13lU1bKlqmihEyZDcYCnv4fiv4WUKFELNRXucDgm1d9M)
+
+## 2. Use Case
+TODO
+
+## 3. Proposed Commands
+### 3.1 RAG Ingestion Pipeline Command
```

Comment:
The CLI integration team might be against modifying the CLI in version 1.4. should also offer ingesting triggering from the config file only

Reply:
Do you want to suggest some shortcuts to run the pipeline(s) without the command?
This can work only for the `chat` command, but there is no `ingest` command to configure ATM: maybe we can provide a notebook in a different repo to manually apply the same pipeline, WDYT?

Reply:
notebook  can work or make command